test-framework/junit5-internal/src/main/java/io/quarkus/test/QuarkusDevModeTest.java

```diff
@@ -539,11 +539,13 @@ public void modifyTestSourceFile(Class<?> sourceFile, Function<String, String> m
      * @param sourceFile
      */
     public void addSourceFile(Class<?> sourceFile) {
-        Path path = copySourceFilesForClass(projectSourceRoot, deploymentSourcePath, testLocation,
+        Path path = findTargetSourceFilesForPath(projectSourceRoot, deploymentSourcePath, testLocation,
+                testLocation.resolve(sourceFile.getName().replace(".", "/") + ".class"));
+        long old = modTime(path.getParent());
```

Comment:
`findTargetSourceFilesForPath` can return `null` and you don't test it AFAICS. Maybe it's a corner case but it seems to be an open door for NPEs.

Reply:
The original copySourceFilesForClass could also return null. I will make them throw an exception instead.